### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by Reinmir**

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -1,19 +1,53 @@
...
+import { RoutePath } from "../RoutePath/RoutePath";
+
 import "./App.css";
 
 const App = () => {
+  const { isAuth } = useAppSelector((state) => state.user);
+  const { setUserData } = UseActions();
+
+  useEffect(() => {
+    const User = localStorage.getItem(LocalStorageKey.authKey);
```
Comment (IgorKravtsov) [inline src/components/App/App.tsx]: Please do not use capital letters for variables

File: src/components/FormBuilder/FormBuilder.tsx [hunk 2]
```diff
@@ -9,31 +9,39 @@ import PasswordInput from "../PasswordInput/PasswordInput";
...
 
+import "./style.scss";
+
 interface FormBuilderProps {
   config: ConfigProps[];
   setInputValues?: Function;
   aboveLink?: React.ReactNode;
-  label: string;
+  label?: string;
+  incorrect?: React.ReactNode;
```
Comment (IgorKravtsov) [inline src/components/FormBuilder/FormBuilder.tsx]: Please rename it to "incorrectLabel" or smth like that

File: src/components/Input/Input.tsx [hunk 3]
```diff
@@ -13,6 +13,7 @@ interface InputProps extends ConfigProps {
   onChange?: any;
```
Comment (IgorKravtsov) [inline src/components/Input/Input.tsx]: Please don't use any type

File: src/components/PasswordInput/PasswordInput.tsx [hunk 4]
```diff
@@ -7,9 +7,11 @@ import { invisibleIcon, visibleIcon } from "../Input/inputIcons";
 import InputConfigsProps from "src/types/InputConfigs";
 
 interface PasswordInputProps extends ConfigProps {
-  onChange: Function;
-  onBlur: Function;
-  error: any;
+  onChange?: Function;
```
Comment (IgorKravtsov) [inline src/components/PasswordInput/PasswordInput.tsx]: Please don't use Fuction type if it is possible

File: src/enums/localStorageEnum.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export enum LocalStorageKey {
+  authKey = "auth",
```
Comment (IgorKravtsov) [inline src/enums/localStorageEnum.ts]: actually you don't need to use ```authKey``` for enum ```LocalStorageKey```
It is a weird to read ```LocalStorageKey.authKey``` better solution ```LocalStorageKey.Auth``` or smth like that

File: src/hooks/useInput.ts [hunk 6]
```diff
@@ -27,8 +27,7 @@ export const useInput = ({ config, setInputValues }: useInputProps) => {
 
   const getError = (
     value: string,
-    validations: ValidationsProps[] | undefined = [],
-    index: number
+    validations: ValidationsProps[] | undefined = []
```
Comment (IgorKravtsov) [inline src/hooks/useInput.ts]: Please siplify to validations?: ValidationsProps[]

File: src/index.tsx [hunk 7]
```diff
@@ -1,18 +1,21 @@
...
 const root = ReactDOM.createRoot(
   document.getElementById("root") as HTMLElement
 );
 root.render(
-  <React.StrictMode>
-    <Router>
-      <App />
-    </Router>
-  </React.StrictMode>
+  <Provider store={store}>
```
Comment (IgorKravtsov) [inline src/index.tsx]: Please wrap <Provider> inside <React.StrictMode>

File: src/pages/Login/Login.tsx [hunk 8]
```diff
@@ -1,32 +1,91 @@
...
+  const navigate = useNavigate();
+
+  const handleSubmit = (items: FormBuilderReturnType[]) => {
+    const userLocalStorage = localStorage.getItem(LocalStorageKey.authKey);
+    const authStorage = userLocalStorage ? JSON.parse(userLocalStorage) : null;
+    const authInput = items.reduce((prev, curr) => {
+      const key = curr.name as keyof UserInput;
+      return (prev[key] = curr.value), prev;
+    }, {} as UserInput);
+    console.log(JSON.stringify(authStorage));
```
Comment (IgorKravtsov) [inline src/pages/Login/Login.tsx]: Please remove all console.logs

File: src/pages/MainPage/MainPage.tsx [hunk 9]
```diff
@@ -1,12 +1,19 @@
...
+import "./style.scss";
 
 const MainPage = () => {
-
-
-
-  return <div></div>;
+  return (
+    <>  
+    <Header></Header>
```
Comment (IgorKravtsov) [inline src/pages/MainPage/MainPage.tsx]: Please use self-closing tag if you don't have children

File: src/pages/MainPage/MainPage.tsx [hunk 10]
```diff
@@ -1,12 +1,19 @@
...
+      <div className="mainpage__container">
+        <div className="mainpage__wrapper">
+          <div></div>
```
Comment (IgorKravtsov) [inline src/pages/MainPage/MainPage.tsx:12]: Please use self-closing tag if you don't have children

File: src/pages/MainPage/style.scss [hunk 11]
```diff
@@ -0,0 +1,12 @@
+.mainpage {
+  &__container {
+    background: url(../../assests/img/Background-mainpage.png) no-repeat center
+      center fixed;
+    background-size: cover;
+    min-height: 100vh;
+  }
+  &__wrapper{
+    min-height: 100vh;
+    min-width: 1000wh;
```
Comment (IgorKravtsov) [inline src/pages/MainPage/style.scss]: min-width 100wh?
I guess it's wrong

File: src/pages/Registration/Registration.tsx [hunk 12]
```diff
@@ -1,27 +1,62 @@
...
+import { UserInput } from "src/types/UserInput";
 
 import config from "./config";
 
 import "./style.scss";
 
 const Registration = () => {
+  const navigate = useNavigate();
+  const user = useAppSelector((state) => state.user);
+  console.log(user);
```
Comment (IgorKravtsov) [inline src/pages/Registration/Registration.tsx]: remove console.log
Please check in all places

**Review by IgorKravtsov — CHANGES_REQUESTED** [review]

File: src/components/App/App.tsx (hunk 1 above)
Comment (Reinmir) [reply src/components/App/App.tsx]: Done.

File: src/components/FormBuilder/FormBuilder.tsx (hunk 2 above)
Comment (Reinmir) [reply src/components/FormBuilder/FormBuilder.tsx]: Done.

File: src/enums/localStorageEnum.ts (hunk 5 above)
Comment (Reinmir) [reply src/enums/localStorageEnum.ts]: Done.

File: src/components/PasswordInput/PasswordInput.tsx (hunk 4 above)
Comment (Reinmir) [reply src/components/PasswordInput/PasswordInput.tsx]: Done.

File: src/components/Input/Input.tsx (hunk 3 above)
Comment (Reinmir) [reply src/components/Input/Input.tsx]: Done.

File: src/index.tsx (hunk 7 above)
Comment (Reinmir) [reply src/index.tsx]: Done.

File: src/pages/MainPage/MainPage.tsx (hunk 9 above)
Comment (Reinmir) [reply src/pages/MainPage/MainPage.tsx]: Done.

File: src/pages/MainPage/style.scss (hunk 11 above)
Comment (Reinmir) [reply src/pages/MainPage/style.scss]: Fixed.

File: src/pages/Registration/Registration.tsx (hunk 12 above)
Comment (Reinmir) [reply src/pages/Registration/Registration.tsx]: Removed.

File: src/pages/MainPage/MainPage.tsx (hunk 10 above)
Comment (Reinmir) [reply src/pages/MainPage/MainPage.tsx:12]: Done.

**Pull request merged**